packages/nimble-components/src/select/tests/select.spec.ts

```diff
@@ -831,413 +843,462 @@ describe('Select', () => {
-            const option = element.options[1] as ListOption;
-            await pageObject.openAndSetFilterText('Three');
-            expect(option.visuallyHidden).toBeTrue();
+                expect(option.visuallyHidden).toBeFalse();
+            });
 
-            element.removeChild(option);
-            await waitForUpdatesAsync();
-            element.appendChild(option);
-            await waitForUpdatesAsync();
+            it('option that is not hidden but is visuallyHidden, if removed and then added back, will be visible', async () => {
+                const option = element.options[0]! as ListOption;
+                option.visuallyHidden = true;
+                element.removeChild(option);
+                await waitForUpdatesAsync();
+                element.appendChild(option);
+                await waitForUpdatesAsync();
 
-            expect(option.visuallyHidden).toBeTrue();
+                expect(option.visuallyHidden).toBeFalse();
+            });
         });
 
-        it('option that is not hidden but visuallyHidden, if removed from DOM, and then re-added to DOM while dropdown closed, is visible in dropdown when opened', async () => {
-            const option = element.options[1] as ListOption;
-            await pageObject.openAndSetFilterText('Three');
-            expect(option.visuallyHidden).toBeTrue();
+        describe('with filterMode = manual', () => {
+            beforeEach(async () => {
+                ({ element, connect, disconnect } = await setup());
+                element.style.width = '200px';
+                element.filterMode = FilterMode.manual;
+                await connect();
+                pageObject = new SelectPageObject(element);
+            });
 
-            element.removeChild(option);
-            await waitForUpdatesAsync();
-            await pageObject.clickAway(); // clears filter
-            element.appendChild(option);
-            await waitForUpdatesAsync();
-            pageObject.clickSelect();
+            afterEach(async () => {
+                await disconnect();
+            });
+
+            it('does not automatically filter options when filterMode is manual', async () => {
+                await pageObject.openAndSetFilterText('o');
+                const filteredOptions = pageObject.getFilteredOptions();
+                expect(filteredOptions.length).toBe(6);
+            });
+
+            it('emits filter-input event when filter text is entered', async () => {
+                const filterInputEvent = jasmine.createSpy();
+                element.addEventListener('filter-input', filterInputEvent);
+                await pageObject.openAndSetFilterText('o');
+                expect(filterInputEvent).toHaveBeenCalledTimes(1);
+            });
 
-            expect(option.visuallyHidden).toBeFalse();
+            it('emits filter-input event with empty filterText when dropdown is closed', async () => {
```

Comment:
Add test case that event is emitted when opening dropdown?

Reply:
We don't emit event on dropdown open. In the `openChanged` implementation you will see an early return if the dropdown is open.